### PR TITLE
Resolvers cleanup

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- cleanup after refactor
  
## Proposed changes
- removed the requester logic out from resolvers along with multiple stubs

## Testing procedure
- to be tested when the feat branch is ready

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: debug/factory/interceptorResolverDebuggerFactory.go [hunk 1]
```diff
@@ -5,10 +5,10 @@ import (
...
-// NewInterceptorResolverDebuggerFactory will instantiate an InterceptorResolverDebugHandler based on the provided config
-func NewInterceptorResolverDebuggerFactory(config config.InterceptorResolverDebugConfig) (InterceptorResolverDebugHandler, error) {
+// NewInterceptorResolverDebuggerFactory will instantiate an InterceptorDebugHandler based on the provided config
+func NewInterceptorResolverDebuggerFactory(config config.InterceptorResolverDebugConfig) (InterceptorDebugHandler, error) {
 	if !config.Enabled {
-		return resolver.NewDisabledInterceptorResolver(), nil
+		return resolver.NewDisabledInterceptorDebugHandler(), nil
 	}
 
 	return resolver.NewInterceptorResolver(config)
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply debug/factory/interceptorResolverDebuggerFactory.go:14]: right, renamed to InterceptorDebugHandler

File: debug/factory/interceptorDebuggerFactory_test.go [hunk 2]
```diff
@@ -4,36 +4,36 @@ import (
...
-	"github.com/ElrondNetwork/elrond-go/debug/resolver"
+	"github.com/ElrondNetwork/elrond-go/debug/handler"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestNewInterceptorResolverDebuggerFactory_DisabledShouldWork(t *testing.T) {
 	t.Parallel()
 
-	irdh, err := NewInterceptorResolverDebuggerFactory(
+	irdh, err := NewInterceptorDebuggerFactory(
```
Comment (SebastianMarian) [inline debug/factory/interceptorDebuggerFactory_test.go:14]: idf instead irdh  here and in L28?

File: debug/handler/export_test.go [hunk 3]
```diff
@@ -1,6 +1,6 @@
-package resolver
+package handler
 
-func (ir *interceptorResolver) Events() []*event {
+func (ir *interceptorDebugHandler) Events() []*event {
```
Comment (SebastianMarian) [inline debug/handler/export_test.go:3]: idh instead ir as pointer receivers?

File: debug/handler/interceptorDebugHandler.go [hunk 4]
```diff
@@ -103,14 +103,14 @@ type interceptorResolver struct {
...
+// NewInterceptorDebugHandler creates a new interceptorDebugHandler able to hold requested-intercepted information
+func NewInterceptorDebugHandler(config config.InterceptorResolverDebugConfig) (*interceptorDebugHandler, error) {
 	lruCache, err := cache.NewLRUCache(config.CacheSize)
 	if err != nil {
-		return nil, fmt.Errorf("%w when creating NewInterceptorResolver", err)
+		return nil, fmt.Errorf("%w when creating NewInterceptorDebugHandler", err)
 	}
 
-	ir := &interceptorResolver{
+	ir := &interceptorDebugHandler{
```
Comment (SebastianMarian) [inline debug/handler/interceptorDebugHandler.go:113]: idh here and to all pointer receivers below?

File: debug/handler/interceptorDebugHandler_test.go [hunk 5]
```diff
@@ -35,14 +35,12 @@ func mockTimestampHandler() int64 {
...
 
-//------- NewInterceptorResolver
-
 func TestNewInterceptorResolver_InvalidSizeShouldErr(t *testing.T) {
 	t.Parallel()
 
 	cfg := createWorkableConfig()
 	cfg.CacheSize = -1
-	ir, err := NewInterceptorResolver(cfg)
+	ir, err := NewInterceptorDebugHandler(cfg)
```
Comment (SebastianMarian) [inline debug/handler/interceptorDebugHandler_test.go:43]: idh instead ir here and in the whole file?

File: debug/handler/disabledInterceptorDebugHandler.go [hunk 6]
```diff
@@ -8,7 +8,7 @@ func NewDisabledInterceptorDebugHandler() *disabledInterceptorDebugHandler {
 	return &disabledInterceptorDebugHandler{}
 }
 
-// LogRequestedData dos nothing
+// LogRequestedData does nothing
 func (dir *disabledInterceptorDebugHandler) LogRequestedData(_ string, _ [][]byte, _ int, _ int) {
```
Comment (SebastianMarian) [inline debug/handler/disabledInterceptorDebugHandler.go:12]: didh or idh instead dir to all pointer recievers?

**Review by SebastianMarian — DISMISSED** [review]

**Review by iulianpascalau — COMMENTED** [review]
only what @SebastianMarian said

**Review by iulianpascalau — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**